extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/runtime/util/VersionHandler.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.mongodb.panache.common.runtime.util;
+
+import java.lang.reflect.Field;
+import java.util.stream.Stream;
+
+import io.quarkus.mongodb.panache.common.Version;
+
+/**
+ * Handle @Version in queries.
+ **/
+public class VersionHandler {
+
+    public final Object entity;
+    public final Field versionField;
+    public final Long versionValue;
+
+    private VersionHandler(Object entity, Field versionField, Long versionValue) {
+        this.entity = entity;
+        this.versionField = versionField;
+        this.versionValue = versionValue;
+    }
+
+    public static VersionHandler of(Object entity) {
+        Field versionField = extractVersionField(entity);
+        Long versionValue = extractVersionValue(versionField, entity);
+
+        return new VersionHandler(entity, versionField, versionValue);
+    }
+
+    public boolean containsVersionAnnotation() {
+        return versionField != null;
+    }
+
+    public boolean containsVersionValue() {
+        return versionValue != null;
+    }
+
+    public boolean containsVersionAnnotationAndValue() {
+        return containsVersionAnnotation() && containsVersionValue();
+    }
+
+    private static Field extractVersionField(Object entity) {
+        Field[] fields = entity.getClass().getFields();
+        Field[] declaredFields = entity.getClass().getDeclaredFields();
+        return Stream.of(fields, declaredFields)
+                .flatMap(Stream::of)
+                .filter(field -> field.isAnnotationPresent(Version.class))
+                .findFirst().orElse(null);
+    }
+
+    public static Long extractVersionValue(Field versionField, Object entity) {
+        try {
+            if (versionField == null) {
+                return null;
+            }
+
+            boolean canAccess = versionField.canAccess(entity);
+            versionField.setAccessible(true);
+            Long versionValue = (Long) versionField.get(entity);
+            versionField.setAccessible(canAccess);
```

Comment:
This setting of 'accessible' back and forth is unnecessary.  Calling `setAccessible()` only affects the `Field` instance (in this case) that it was called on. Code else where that extracts a `Field` reflectively gets a different instance so there is no cross-contamination risk here.

Reply:
changed.